tests/function/gtVersion.cpp

```diff
@@ -0,0 +1,145 @@
+ *             defined in config_int.h (and the function returns FPGA_OK)
+ */
+TEST(LibopaecVersionCommonALL, version_02) {
+	fpga_version version = { 0xFF, 0xFF, 0xFFFF };
+
+	EXPECT_EQ(FPGA_OK, fpgaGetOPAECVersion(&version));
+	EXPECT_EQ(INTEL_FPGA_API_VER_MAJOR, version.major);
+	EXPECT_EQ(INTEL_FPGA_API_VER_MINOR, version.minor);
+	EXPECT_EQ(INTEL_FPGA_API_VER_REV, version.patch);
+}
+
+/**
+ * @test       version_03
+ *
+ * @brief      When I retrieve fpga_version information using
+ *             fpgaGetOPAECVersionString using a NULL pointer as the return
+ *             buffer, the function returns FPGA_INVALID_PARAM.
+ */
+TEST(LibopaecVersionCommonALL, version_03) {
+	EXPECT_EQ(FPGA_INVALID_PARAM, fpgaGetOPAECVersionString(NULL, 42));
+}
+
+/**
+ * @test       version_04
+ *
+ * @brief      When I retrieve fpga_version information using
+ *             fpgaGetOPAECVersionString using a size shorter than the minimum
+ *             possible (which is 6) the function returns FPGA_EXCEPTION.
+ */
+TEST(LibopaecVersionCommonALL, version_04) {
+	char str[80];
+
+	EXPECT_EQ(FPGA_EXCEPTION, fpgaGetOPAECVersionString(str, 5));
+}
+
+/**
+ * @test       version_05
+ *
+ * @brief      When I retrieve fpga_version information using
+ *             fpgaGetOPAECVersionString, the returned string represents
+ *             a string comprised of the constants defined in config_int.h
+ *             (and the function returns FPGA_OK)
+ */
+TEST(LibopaecVersionCommonALL, version_05) {
+	char want[80];
+	char have[80];
+
+	snprintf(want, 80, "%d.%d.%d",
+			INTEL_FPGA_API_VER_MAJOR,
+			INTEL_FPGA_API_VER_MINOR,
+			INTEL_FPGA_API_VER_REV
+		);
+
+	EXPECT_EQ(FPGA_OK, fpgaGetOPAECVersionString(have, 80));
+	EXPECT_STREQ(want, have);
+}
+
+/**
+ * @test       version_06
+ *
+ * @brief      When I retrieve fpga_version information using
+ *             fpgaGetOPAECBuildString using a NULL pointer as the return
+ *             buffer, the function returns FPGA_INVALID_PARAM.
+ */
+TEST(LibopaecVersionCommonALL, version_06) {
+	EXPECT_EQ(FPGA_INVALID_PARAM, fpgaGetOPAECBuildString(NULL, 42));
+}
+
+/**
+ * @test       version_07
+ *
+ * @brief      When I retrieve fpga_version information using
+ *             fpgaGetOPAECBuildString using a size of 0
+ *             the function returns FPGA_EXCEPTION.
+ */
+TEST(LibopaecVersionCommonALL, version_07) {
+	char str[80];
+
+	EXPECT_EQ(FPGA_EXCEPTION, fpgaGetOPAECBuildString(str, 0));
+}
+
+/**
+ * @test       version_08
+ *
+ * @brief      When I retrieve fpga_version information using
+ *             fpgaGetOPAECBuildString, the returned string equals
+ *             the constant defined in config_int.h (and the
+ *             function returns FPGA_OK)
+ */
+TEST(LibopaecVersionCommonALL, version_08) {
+	char want[] = INTEL_FPGA_API_HASH;
+	char have[80];
+
+	EXPECT_EQ(FPGA_OK, fpgaGetOPAECBuildString(have, 80));
+	EXPECT_STREQ(want, have);
```

Comment:
can we replace `want` with `"INTEL_FPGA_API_HASH"`?

Reply:
Yes. I did this for consistency between `version_08` and `version_05`.